staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -464,6 +464,10 @@ func (u *Unstructured) GetCreationTimestamp() metav1.Time {
 
 func (u *Unstructured) SetCreationTimestamp(timestamp metav1.Time) {
 	ts, _ := timestamp.MarshalQueryParameter()
+	if len(ts) == 0 {
+		u.setNestedField(nil, "metadata", "creationTimestamp")
```

Comment:
hmm... actually, should this be storing `nil` or removing from the map? @ironcladlou, any thoughts here?

Reply:
I suggested removing nil keys from the map generally in https://github.com/kubernetes/kubernetes/issues/48211, but I wouldn't say there's consensus on that point yet.

@deads2k suggested we rely on `omitempty` in the external types to accommodate a nil here (mirroring his prior counterpoint to https://github.com/kubernetes/kubernetes/issues/48211#issuecomment-311828662), which I'm fine with.

Maybe some more godocs would help here in the meantime?

Reply:
This field is omitempty.

@juanvallejo I prefer to delete the entry, then this lgtm.

Reply:
Thanks, will update